repose-aggregator/components/filters/herp-filter/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala

```diff
@@ -121,8 +123,17 @@ class HerpFilter @Inject()(configurationService: ConfigurationService,
       else stripHeaderParams(allProjectIds.maxBy(getQuality))
     }
 
-    val projectIds = httpServletRequest.getHeaders(OpenStackServiceHeader.TENANT_ID.toString).asScala
-      .++(httpServletRequest.getHeaders(X_PROJECT_ID).asScala).toTraversable
+    def getSplitHeaders(headerInteractor: HeaderInteractor, headers: List[String]): Traversable[String] = {
+      headers.foldLeft(Traversable[String]()) { (accumulator, current) =>
+        accumulator.++(headerInteractor.getSplittableHeaders(current).asScala)
+      }
+    }
+
+    val tenantProjectHeaders = List(OpenStackServiceHeader.TENANT_ID.toString, X_PROJECT_ID)
```

Comment:
Also, why a `List` (here and in the test)? If anything, I would think of this as a `Set`. `List` seems odd because it's a specific collection type -- not like `Traversable` -- which makes me think there must be a reason for using it... but I'm not sure there is?